components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1812,125 +1816,334 @@ private int handleStartIndexEqualsNULL(Integer startIndex) {
+     */
+    private List<Object> filterGroups(Node rootNode, int startIndex, Integer count, String sortBy, String sortOrder,
+            String domainName, Map<String, Boolean> requiredAttributes)
+            throws NotImplementedException, CharonException {
+
+        // Handle count equals NULL scenario.
+        count = handleLimitEqualsNULL(count);
+        if (rootNode instanceof ExpressionNode) {
+            return filterGroupsBySingleAttribute((ExpressionNode) rootNode, startIndex, count, sortBy, sortOrder,
+                    domainName, requiredAttributes);
+        } else if (rootNode instanceof OperationNode) {
+            String error = "Complex filters are not supported yet";
             throw new NotImplementedException(error);
+        } else {
+            throw new CharonException("Unknown operation. Not either an expression node or an operation node.");
         }
+    }
 
-        if (isFilterUsersAndGroupsOnlyFromPrimaryDomainEnabled()) {
-            attributeValue = prependDomain(attributeValue);
-        } else if (isFilteringEnhancementsEnabled()) {
-            if (SCIMCommonConstants.EQ.equalsIgnoreCase(filterOperation)) {
-                if (StringUtils.equals(attributeName, SCIMConstants.GroupSchemaConstants.DISPLAY_NAME_URI)) {
-                    attributeValue = prependDomain(attributeValue);
-                }
-            }
-        }
+    /**
+     * Filter groups with a single attribute.
+     *
+     * @param node
+     * @param startIndex
+     * @param count
+     * @param sortBy
+     * @param sortOrder
+     * @param domainName
+     * @param requiredAttributes
+     * @return
+     * @throws CharonException
+     */
+    private List<Object> filterGroupsBySingleAttribute(ExpressionNode node, int startIndex, int count, String sortBy,
+            String sortOrder, String domainName, Map<String, Boolean> requiredAttributes) throws CharonException {
 
+        String attributeName = node.getAttributeValue();
+        String filterOperation = node.getOperation();
+        String attributeValue = node.getValue();
         if (log.isDebugEnabled()) {
-            log.debug("Listing groups with filter: " + attributeName + filterOperation +
-                    attributeValue);
+            log.debug("Filtering groups with filter: " + attributeName + " + " + filterOperation + " + "
+                    + attributeValue);
+        }
+        // Check whether the filter operation is supported for filtering in groups.
+        if (isFilteringNotSupported(filterOperation)){
+            String errorMessage = "Filter operation: " + filterOperation + " is not supported for groups filtering.";
+            throw new CharonException(errorMessage);
         }
+        // Resolve the domain name in request according to isFilterUsersAndGroupsOnlyFromPrimaryDomainEnabled() or
+        // isFilteringEnhancementsEnabled() or domain name embedded in the filter attribute value.
```

Comment:
Rather can we mention the properties

Reply:
Fixed: 788feb46892fd4c42062f6197f51d104401db991